components/liquidity/LiquidityModal.tsx

```diff
@@ -93,24 +91,33 @@ const LiquidityModal = ({
 
           const userBalance = isBaseAsset
             ? userBaseBalance
-            : new Decimal(userAssetBalances[index].free.toString());
+            : new Decimal(
+                userAssetBalances == null
+                  ? 0
+                  : userAssetBalances[index].free.toString(),
+              );
           const poolBalance = isBaseAsset
             ? new Decimal(poolBaseBalance.toString())
-            : new Decimal(poolAssetBalances[index].free.toString());
+            : new Decimal(
+                poolAssetBalances == null
+                  ? 0
+                  : poolAssetBalances[index].free.toString(),
+              );
 
-          const id = assetObjStringToId(weight.assetId);
+          const id = assetObjStringToId(weight!.assetId);
```

Comment:
I think this is ok for now but in the future I rather fix this by changing the sdk types, I don't think it makes sense for a weight to be `null`, if it doesn't exist it should be in the array